### PR TITLE
NIAD-754: Create initial Spring Boot project structure

### PR DESCRIPTION
**Pull request by Broom94**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,28 +26,33 @@ This value can be overriden using `SCR_LOGGING_FORMAT` environment variable.
 Alternatively, an external `logback.xml` with much more customizations can be provided using `-Dlogback.configurationFile` JVM parameter.
```
Comment (Broom94) [inline README.md:26]: Need to add logback and update this part of readme

File: docker/service/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FROM adoptopenjdk/openjdk14:slim
+
+COPY gp2gp.jar /
+
+ENTRYPOINT java -jar /gp2gp.jar --port 9000
```
Comment (Broom94) [inline docker/service/Dockerfile:5]: update port

File: docker/service/build-service-image.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
```
Comment (Broom94) [inline docker/service/build-service-image.sh:1]: remove?

File: docker/start-local-environment.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
```
Comment (Broom94) [inline docker/start-local-environment.sh:1]: remove?

File: service/src/intTest/resources/application.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+server:
+  port: ${GP2GP_SERVER_PORT:9000}
+
+management:
+  endpoints:
+    web:
+      base-path: /
+      path-mapping.health: healthcheck
+      exposure:
+        include: info, health, metrics, mappings
```
Comment (Broom94) [inline service/src/intTest/resources/application.yml:10]: change this

File: docker/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+version: '3'
+
+services:
+  gp2gp:
+    image: uk.nhs/gp2gp:0.0.1-SNAPSHOT
+    ports:
+      - "8080"
```
Comment (Broom94) [inline docker/docker-compose.yml:7]: 8080:8080?

File: README.md [hunk 7]
```diff
@@ -26,28 +26,33 @@ This value can be overriden using `SCR_LOGGING_FORMAT` environment variable.
...
-TODO
-<!-- * Navigate to `service`
-* Run: `./gradlew check` -->
+* Navigate to `service`
+* Run: `./gradlew check` 
+
+### Licensing
+This code is dual licensed under the MIT license and the OGL (Open Government License). Any new work added to this repository must conform to the conditions of these licenses. In particular this means that this project may not depend on GPL-licensed or AGPL-licensed libraries, as these would violate the terms of those libraries' licenses.
+
+The contents of this repository are protected by Crown Copyright (C).
```
Comment (Broom94) [inline README.md:58]: should we have pre-requests, i.e. install docker, lombok plugin?

File: docker/service/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,5 @@
+FROM adoptopenjdk/openjdk14:slim
```
Comment (mattd-kainos) [inline docker/service/Dockerfile:1]: use jdk11 which is the latest LTS. jdk14 is already out of support

File: README.md (hunk 1 above)
Comment (mattd-kainos) [reply README.md:26]: Can just copy what was used in SCR or 111?

**Review by mattd-kainos — CHANGES_REQUESTED** [review]
jdk version

File: service/src/intTest/resources/application.yml [hunk 9]
```diff
@@ -0,0 +1,10 @@
+server:
```
Comment (mattd-kainos) [inline service/src/intTest/resources/application.yml:1]: Consider adding an integration test for healthcheck? That way there is at least something in this intTest src dir.

File: .gitignore [hunk 10]
```diff
@@ -1 +1,34 @@
...
+hs_err_pid*
+
+.idea/
+*.iml
+*/.gradle/
+*.bin
+*/build/*
+*.DS_Store
+
+service/out/*
```
Comment (mattd-kainos) [inline .gitignore:34]: Could we also add

```
.project
.settings
```

I'm using VS Code! :gasp:

File: README.md [hunk 11]
```diff
@@ -20,34 +20,39 @@ Variables without a default value and not marked optional, *MUST* be defined for
...
-This value can be overriden using `SCR_LOGGING_FORMAT` environment variable.
+This value can be overriden using `GP2GP_LOGGING_FORMAT` environment variable.
 Alternatively, an external `logback.xml` with much more customizations can be provided using `-Dlogback.configurationFile` JVM parameter.
 
 ## How to run service:
-TODO
-<!-- * Navigate to `ddocker/service`
+* Navigate to `docker/service`
 * Add environment tag `export TAG=latest`
 * Run script: `build-image.sh` (excute privileges might be required `chmod +x build-image.sh`)
```
Comment (mattd-kainos) [inline README.md:31]: Should this be `build-service-image.sh`?

Comment (mattd-kainos) [reply README.md:31]: Could probably just skip this step - `start-local-environment.sh` calls this script anyway

File: README.md (hunk 7 above)
Comment (mattd-kainos) [reply README.md:57]: I think what we have is fine for now.

File: docker/service/Dockerfile [hunk 12]
```diff
@@ -0,0 +1,5 @@
+FROM adoptopenjdk/openjdk11:slim
+
+COPY gp2gp.jar /
+
+ENTRYPOINT java -jar /gp2gp.jar --port 8080
```
Comment (mattd-kainos) [inline docker/service/Dockerfile:5]: I suspect that --port 8080 doesn't actually do anything here

**Review by mattd-kainos — APPROVED** [review]

**Pull request merged**